### DISCUSSION TITLE
Porting to newer versions (107, 108, etc)

### DISCUSSION THREAD

**Discussion by chirayudesai** [bromite/bromite#2423] [Builds]
Starting this to further discuss ways to make the porting process easier, as discussed in #2404

**Comment by chirayudesai** [comment]
Here is the process I use to port to new versions using git

tl;dr
`git cherry-pick $OLD_TAG...branch-with-patches-applied-to-old-tag`

`export OLD_TAG=106.0.5249.163 NEW_TAG=107.0.5304.36`

1. Setup previous, known working patches
```
cd chromium/src
git checkout -b bromite-${OLD_TAG} ${OLD_TAG}
for i in `cat ../bromite/build/bromite_patches_list.txt`; do git am ../bromite/build/patches/$i; if [ $? -ne 0 ]; then echo "Waiting for conflict resolution"; read; fi; done
```

2. Optionally, make sure the above builds. Usually I have the old branch locally anyway

3. Start picking to new version
```
git checkout -b bromite-${NEW_TAG} ${NEW_TAG}
gclient sync -D; gclient runhooks # Optional, just to get it all in place early
git cherry-pick ${OLD_TAG}...bromite-${OLD_TAG}
```

That last command is what does it all. That way we get cherry-pick's better version of patch application compared to `git am`. It handles trivial conflicts easily in my experience (although not all of them)

Then, I use sublime merge with a 3 way diff to resolve the conflicts during step 3.

Finally, I have a window open with patches applied to the old version so I can see the actual diff with context rather than just patch files, to make it easier to resolve conflicts.
